src/NUnitFramework/tests/Constraints/ThrowsExceptionConstraintTests.cs

```diff
@@ -65,19 +66,39 @@ public void SucceedsWithNonVoidReturningFunction()
         };
 
 #if ASYNC
+        private static async Task YieldAsync()
+        {
+#if NET40
+            await TaskEx.Yield();
+#else
+            await Task.Yield();
+#endif
+        }
+
+
         [Test]
         public static void CatchesAsyncException()
         {
             Assert.That(async () =>
             {
-#if NET40
-                await TaskEx.Yield();
-#else
-                await Task.Yield();
-#endif
+                await YieldAsync();
                 throw new Exception();
             }, Throws.Exception);
         }
+        
+        [Test]
+        public static void CatchesAsyncOfTException()
+        {
+            Assert.That(async () =>
+            {
+#pragma warning disable CS0162
+                await YieldAsync();
+                throw new Exception();
+                return 2;
+#pragma warning restore CS0162
+            }, Throws.Exception);
+        }
+
```

Comment:
(While you're changing stuff, there wasn't a blank line here before.)

Reply:
Removed 👍